### ISSUE TITLE
Documentation claims shaprpy can explain Keras models, but code claims it cannot

### ISSUE THREAD

**Issue by mdhimes** [NorskRegnesentral/shapr#394]
Hello.  I am using the Python interface, shaprpy.  The documentation for it claims that
"shaprpy knows how to explain predictions from models from sklearn, keras and xgboost. For other models, one can provide a custom predict_model function (and optionally a custom get_model_specs) to shaprpy.explain."

However, when using a Keras model, the following warning arises:
".../shapr/python/shaprpy/explain.py:255: UserWarning: No pre-built get_model_specs for model of type <class 'keras.src.engine.functional.Functional'>, disabling checks.
  warnings.warn(f'No pre-built get_model_specs for model of type {type(model)}, disabling checks.')
R[write to console]: Note: You passed a model to explain() which is not natively supported, and did not supply a 'get_model_specs' function to explain().
Consistency checks between model and data is therefore disabled."

This quite plainly states that a Keras model "is not natively supported", which seems at odds with the documentation.  Looking in the code seems to confirm this:
https://github.com/NorskRegnesentral/shapr/blob/master/python/shaprpy/explain.py#L313
Note that only sklearn and xgboost are included in this function, but not Keras.

Is this expected behavior?  If so, the warning should be suppressed for this case.  If not, the documentation should be updated to remove mention of native support for Keras models.

**Comment by martinju** [comment]
shaprpy supports (at least some types of) keras models. Here is the code implementing it: 
https://github.com/NorskRegnesentral/shapr/blob/fcae7581597b56a99fe1fcc7d2b71f5936c4f4eb/python/shaprpy/explain.py#L551

And here is an example showing the usage: 
https://github.com/NorskRegnesentral/shapr/blob/master/python/examples/keras_classifier.py

**Comment by mdhimes** [comment]
I don't think this issue should have been closed quite yet.  The example provided shows that shaprpy supports Sequential Keras models, but my experience shows it does not work for Functional Keras models (I can provide a MWE if needed).  The Keras Functional API allows for more complex architectures (e.g., U-Net) and, in my observation, is more widely used than the Sequential API.  

Given the above, I think it is misleading to say that shaprpy supports Keras models without any qualifiers.  I would suggest that the documentation is updated to specify that Sequential Keras models are supported, but that Functional models are not.  I would love to see the code expanded to support Functional Keras models, but I don't have the time to do that myself at present.

**Comment by martinju** [comment]
I added (sequential) to the description, and hope that is OK for now. 
Unfortunately, I don't work with keras myself, so it will be tedious to add such functionality for me. 
We will welcome PRs to do this. 
Note, however, that the package is model-agnostic. For any model outside of keras (sequential), xgboost and sklearn you can just pass the prediction function to explain() yourself.

**Comment by mdhimes** [comment]
Thanks!

I will try to carve out some time to do a detailed investigation.  It's possible that the Functional model works fine and the warning can be ignored.  Not sure when I will get around to this, but I'll follow up if/once I do.
